sync/src/blocks.rs

```diff
@@ -366,23 +370,26 @@ impl BlockCollection {
 			let receipts = UntrustedRlp::new(&r);
 			ordered_trie_root(receipts.iter().map(|r| r.as_raw().to_vec())) //TODO: get rid of vectors here
 		};
-		match self.receipt_ids.get(&receipt_root).cloned() {
-			Some(h) => {
-				self.receipt_ids.remove(&receipt_root);
+		match self.receipt_ids.entry(receipt_root) {
+			Entry::Occupied(mut entry) => {
+				let h = entry.get_mut().pop().expect("Empty vectors are not allowed in insert_receipt; qed");
```

Comment:
It's not that simple. Every response must be matched against a request, otherwise we can't tell good behaviour from bad. I've made an additional commit that tracks unique receipts and only issues one download request for each receipt root.